Assets/BossRoom/Scripts/Shared/ApplicationController.cs

```diff
@@ -52,6 +53,9 @@ private void Awake()
             //this message channel is essential and persists for the lifetime of the lobby and relay services
             scope.BindMessageChannelInstance<ConnectStatus>();
 
+            //this message channel is essential and persists for the lifetime of the lobby and relay services
+            scope.BindMessageChannelInstance<MessageFeedHandler.MessageFeed>();
```

Comment:
So this is a trap of message feeds we shouldn't fall into.
There should be a message queue for in game events, there should be one for connection events. And the feed UI should hook into those different message queues (just like we could have some graphic updates from those same events). We shouldn't have one big message queue with a bunch of different message types in it.

Reply:
This way those queues can be reused by other consumers that are not the feed UI

Reply:
I see. I went that way mainly to solve two constraints. First, a lot of these messages are published locally on the different clients and the server, but here we want to replicate all these messages to all clients. Second, I wanted to keep a history of messages that would live throughout the application that each instance of UIMessageFeed could listen to. So here, this message channel is used to publish the feed history that is constructed client-side from messages created on the server and sent through rpcs. The actual messages are sent through MessageFeedHandler as the singleton pattern.

I'll rework this and look into replacing my singleton use with DI and create a new message channel for in-game messages. I'll also try to think of a different solution to solve those two constraints without a specific message channel to publish the feed history

Reply:
I see.
We might need a new message channel type then. This would be a great use for a networked message channel, that'd distribute your messages not only locally but to clients as well, this could be a great use case for this.
cc @LukeStampfli @pdeschain we already discussed this

Reply:
Why do you want to keep a history, not sure I understood that part.

Reply:
I figured that we could want to look at older messages, like for examples in some games you can open the game feed and look at older chat messages or events like the kill log in first person shooters

Reply:
If we don't want this though, I can make this much simpler and only keep the history in UIMessageFeed which would be cleared after each scene switch

Reply:
I'm not sure what the point of having a buffered message channel would be here. What makes the latest message interesting, but not those before it?

Reply:
I see, but if the point is to not send the whole chat history and instead only publish messages as they are created, what would be the state that we want to get?

Reply:
sorry, when I mentioned buffered above, that just generally speaking. I wasn't suggesting we use it in this case here.

Reply:
oh my bad I misunderstood your point